include/dca/phys/models/analytic_hamiltonians/threeband_hubbard.hpp

```diff
@@ -0,0 +1,288 @@
+      const parameters_type& parameters);
+
+  template <class domain>
+  static void initialize_H_symmetry(func::function<int, domain>& H_symmetry);
+
+  // Initializes the tight-binding (non-interacting) part of the momentum space Hamiltonian.
+  // Preconditions: The elements of KDmn are two-dimensional (access through index 0 and 1).
+  template <typename ParametersType, typename ScalarType, typename BandDmn, typename SpinDmn, typename KDmn>
+  static void initialize_H_0(
+      const ParametersType& parameters,
+      func::function<ScalarType, func::dmn_variadic<func::dmn_variadic<BandDmn, SpinDmn>,
+                                                    func::dmn_variadic<BandDmn, SpinDmn>, KDmn>>& H_0);
+};
+
+template <typename point_group_type>
+int threeband_hubbard<point_group_type>::transformationSignofR(int b1, int b2, int s) {
+  if (!std::is_same<point_group_type, domains::D4>::value)
+    return 1;
+
+  if (b1 == b2)
+    return 1;
+  else if (b1 != 0 && b2 != 0){
+    if (s == 0 || s == 6)
+      return 1;
+    else
+      return 0;
+  }
+  else if ((b1 != b2 && b1 == 0)||(b1 != b2 && b2 == 0)) {
+    if (s == 0)     
+      return 1;
+    else if (s == 6)
+      return -1;
+    else
+      return 0;
+  }
+
+}
+
+template <typename point_group_type>
+int threeband_hubbard<point_group_type>::transformationSignofK(int b1, int b2, int s) {
+  if (!std::is_same<point_group_type, domains::D4>::value)
+    return 1;
+
+  if ((b1 == b2) || (b1 != 0 && b2 != 0))
+    return 1;
+  else // if ((b1 != b2 && b1 == 0)||(b1 != b2 && b2 == 0))
+    return (s == 0 || s == 2 || s == 5 || s == 7) ? 1 : -1;
+    //if (s == 0 || s == 2 || s == 5 || s == 7)
+    //  return 1;
+    // else // if (s == 1 || s == 3 || s == 4 || s == 6)
+    //  return -1;
+  
+
+}
+
+
+template <typename point_group_type>
+double* threeband_hubbard<point_group_type>::initialize_r_DCA_basis() {
+  static double* r_DCA = new double[4];
+  r_DCA[0] = 1.0;
+  r_DCA[1] = 0.0;
+  r_DCA[2] = 0.0;
+  r_DCA[3] = 1.0;
+
+  return r_DCA;
+}
+template <typename point_group_type>
+double* threeband_hubbard<point_group_type>::initialize_k_DCA_basis() {
+  static double* k_DCA = new double[4];
+
+  k_DCA[0] = 2 * M_PI;
+  k_DCA[1] = 0.;
+  k_DCA[2] = 0.;
+  k_DCA[3] = 2 * M_PI;
+
+  return k_DCA;
+}
+
+template <typename point_group_type>
+double* threeband_hubbard<point_group_type>::initialize_r_LDA_basis() {
+  static double* r_LDA = new double[4];
+
+  r_LDA[0] = 1.;
+  r_LDA[1] = 0.;
+  r_LDA[2] = 0.;
+  r_LDA[3] = 1.;
+
+  return r_LDA;
+}
+
+template <typename point_group_type>
+double* threeband_hubbard<point_group_type>::initialize_k_LDA_basis() {
+  static double* k_LDA = new double[4];
+
+  k_LDA[0] = 2. * M_PI;
+  k_LDA[1] = 0.;
+  k_LDA[2] = 0.;
+  k_LDA[3] = 2. * M_PI;
+
+  return k_LDA;
+}
+
+template <typename point_group_type>
+std::vector<int> threeband_hubbard<point_group_type>::get_flavors() {
```

Comment:
this a weird pattern. should not be get_ as it is not a simple getter or setter flavors is in fact not even a member variable.  Why is it important to have just one of this very small object? Why should it be global?  It looks like it should be a member variable. Are we just following a questionable convention an earlier developer used?

```C++
const std::vector<int> flavors{0,1,1};
```

perhaps static if you insist.

Reply:
this is again part of the old model interface. Will see if we can do better.

Reply:
changed to brace list initialization in ad2485498f0d2e09b41105fb887dedbeb9e90280